modules/example-browser/src/caching_materials_manager.ts

```diff
@@ -0,0 +1,201 @@
+   * You need kms:GenerateDataKey permission on the CMK in generatorKeyId.
+   */
+  const generatorKeyId = 'arn:aws:kms:us-west-2:658956600833:alias/EncryptDecrypt'
+
+  /* Adding alternate KMS keys that can decrypt.
+   * Access to kms:Encrypt is required for every CMK in keyIds.
+   * You might list several keys in different AWS Regions.
+   * This allows you to decrypt the data in any of the represented Regions.
+   * In this example, I am using the same CMK.
+   * This is *only* to demonstrate how the CMK ARNs are configured.
+   */
+  const keyIds = ['arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7f']
+
+  /* Need a client provider that will inject correct credentials.
+   * The credentials here are injected by webpack from your environment bundle is created
+   * The credential values are pulled using @aws-sdk/credential-provider-node.
+   * See kms.webpack.config
+   * You should inject your credential into the browser in a secure manner,
+   * that works with your application.
+   */
+  const { accessKeyId, secretAccessKey, sessionToken } = credentials
+
+  /* getClient takes a KMS client constructor
+   * and optional configuration values.
+   * The credentials can be injected here,
+   * because browser do not have a standard credential discover process the way Node.js does.
+   */
+  const clientProvider = getClient(KMS, {
+    credentials: {
+      accessKeyId,
+      secretAccessKey,
+      sessionToken
+    }
+  })
+
+  /* The KMS keyring must be configured with the desired CMKs */
+  const keyring = new KmsKeyringBrowser({ clientProvider, generatorKeyId, keyIds })
+
+  /* Create a cache to hold the material.
+   * In this case we use the local cache provided by the Encryption SDK.
+   * The number is the maximum number of entries that will be cached.
+   * Both encrypt and decrypt requests count independently towards this total.
+   * Elements will be actively removed from the cache.
+   * The default frequency is to check one item every minute.
+   * This can be configure by passing a `proactiveFrequency`
+   * as the second paramter to however often you want to check in milliseconds.
+   */
+  const capacity = 100
+  const cache = getLocalCryptographicMaterialsCache(capacity)
+
+  /* The partition name lets multiple caching CMMs share the same local cryptographic cache.
+   * If you want these CMMs to all cache the same items,
+   * make the partition name the same.
+   * If no partition is supplied a random one will be generated.
+   * This is so that sharing elements in the cache MUST be an intentional operation.
+   */
+  const partition = 'local partition name'
+
+  /* maxAge is the time in milliseconds that an entry will be cached.
+   * Elements are actively removed from the cache.
+   */
+  const maxAge = 1000 * 60
+
+  /* The maximum amount of bytes that will be encrypted under a single data key.
+   * This value is optional,
+   * but you should configure the lowest value possible.
```

Comment:
```suggestion
   * but you should configure the lowest practical value.
```
The lowest possible value is 0 or minInt or whatever. That's not what we mean.